### PR TITLE
hadoop wrapper additions for HA support

### PR DESCRIPTION
**Pull request by duddun**
New ha_init.rb and additions to default attribute file for HA
New hadoop_hdfs_journalnode_init.rb for adding init wrapper cookbook for journal nodes.

* ha_init.rb is needed to have init for HA.
* hadoop_hdfs_journalnode_init is needed to have init for journalnodes.
* New attribute file HA section has mostly placeholders and includes generic values for now (e.g. type of host<number>_fqdn, paths, etc). These will be changed as soon as possible (before any of this is used).

The recipes above will be run from roles that do not exist yet and will be created in a separate pull request.

### CODE REVIEW

**Comment by duddun** [comment]
Moving changes from hadoop_hdfs_namenode_ha_init.rb to hadoop_hdfs_namenode_init.rb and adding new ruby_block entries there.

**Comment by wolf31o2** [comment]
This isn't passing tests. :smile:

File: recipes/hadoop_hdfs_namenode_init.rb [hunk 1]
```diff
@@ -32,3 +32,10 @@
   ### TODO: this should check all dfs name dirs, not just the first
   # only_if { (Dir.entries("#{node['hadoop']['hdfs_site']['dfs.name.dir'].split(',').first}") - %w{ . .. }).empty? }
 end
+
+ruby_block 'master-namenode-init' do
```
Comment (wolf31o2) [inline recipes/hadoop_hdfs_namenode_init.rb]: Label this `initaction-hdfs-namenode-initialize-sharededits`

File: recipes/hadoop_hdfs_namenode_init.rb [hunk 2]
```diff
@@ -32,3 +32,10 @@
...
+  block do
+    resources(:execute => 'hdfs-namenode-initialize-sharededits').run_action(:run)
+  end
+  only_if { node['hadoop']['hdfs_site']['dfs.nameservices'] && node['hadoop']['hdfs_site']["dfs.ha.namenodes.#{clustername}"] && node['hadoop']['hdfs_site']["dfs.ha.namenodes.#{clustername}"].split(',')[0] == node['fqdn']
```
Comment (wolf31o2) [inline recipes/hadoop_hdfs_namenode_init.rb]: Where does `clustername` come from?

File: recipes/hadoop_hdfs_standby_namenode_init.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#
+# Cookbook Name:: hadoop_wrapper
+# Recipe:: hadoop_hdfs_standby_namenode_init
+#
+# Copyright © 2013 Cask Data, Inc.
```
Comment (wolf31o2) [inline recipes/hadoop_hdfs_standby_namenode_init.rb]: Use correct dates...

File: recipes/hadoop_hdfs_standby_namenode_init.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include_recipe 'hadoop_wrapper::default'
+include_recipe 'hadoop::hadoop_hdfs_namenode'
+
+ruby_block 'standby-namenode-init' do
```
Comment (wolf31o2) [inline recipes/hadoop_hdfs_standby_namenode_init.rb]: Label this `initaction-hdfs-namenode-bootstrap-standby`

**Comment by wolf31o2** [comment]
This still needs work and isn't complete. Closing.

**Pull request closed without merging**